core/arch/riscv/kernel/entry.S

```diff
@@ -3,9 +3,165 @@
  * Copyright 2022 NXP
  */
 
-#include <platform_config.h>
+#include <riscv.h>
 #include <asm.S>
+#include <keep.h>
+#include <platform_config.h>
+
+.section .data
+.balign 4
+
+#ifdef CFG_BOOT_SYNC_CPU
+.equ SEM_CPU_READY, 1
+#endif
+
+	/*
+	 * Setup sp to point to the top of the tmp stack for the current CPU:
+	 * sp is assigned stack_tmp_export + cpu_id * stack_tmp_stride
+	 */
+.macro set_sp
+	/* Unsupported CPU, park it before it breaks something */
+	li   t1, CFG_TEE_CORE_NB_CORE
```

Comment:
Please separate the instruction and its arguments with a single tab. Please use this style in all assembly files.